integration/benchmarks_test.go

```diff
@@ -29,112 +28,59 @@ import (
 )
 
 func BenchmarkFind(b *testing.B) {
-	provider := shareddata.BenchmarkSmallDocuments
-
-	b.Run(provider.Name(), func(b *testing.B) {
-		s := setup.SetupWithOpts(b, &setup.SetupOpts{
-			BenchmarkProvider: provider,
-		})
-
-		for name, bc := range map[string]struct {
-			filter bson.D
-		}{
-			"Int32ID": {
-				filter: bson.D{{"_id", int32(42)}},
-			},
-			"Int32One": {
-				filter: bson.D{{"id", int32(42)}},
-			},
-			"Int32Many": {
-				filter: bson.D{{"v", int32(42)}},
-			},
-			"Int32ManyDotNotation": {
-				filter: bson.D{{"v.foo", int32(42)}},
-			},
-		} {
-			b.Run(name, func(b *testing.B) {
-				var firstDocs, docs int
-
-				for range b.N {
-					cursor, err := s.Collection.Find(s.Ctx, bc.filter)
-					require.NoError(b, err)
-
-					docs = 0
-					for cursor.Next(s.Ctx) {
-						docs++
-					}
-
-					require.NoError(b, cursor.Close(s.Ctx))
-					require.NoError(b, cursor.Err())
-					require.Positive(b, docs)
+	for _, provider := range shareddata.AllBenchmarkProviders() {
+		b.Run(provider.Name(), func(b *testing.B) {
+			s := setup.SetupWithOpts(b, &setup.SetupOpts{
+				BenchmarkProvider: provider,
+			})
 
-					if firstDocs == 0 {
-						firstDocs = docs
-					}
+			for name, filter := range map[string]bson.D{
+				"Int32IDIndex":         {{"_id", int32(42)}},
+				"Int32One":             {{"id", int32(42)}},
+				"Int32Many":            {{"v", int32(42)}},
+				"Int32ManyDotNotation": {{"v.foo", int32(42)}},
+			} {
+				if provider == shareddata.BenchSettings && name != "Int32IDIndex" {
```

Comment:
It's great to have additional `BenchSettings` provider! For `BenchSettings` provider, what's the reason for only running for `Int32IDIndex` filter?

Reply:
`id`, `v`, `v.foo` fields are not present in settings documents